cpp/ql/src/Security/CWE/CWE-022/examples/TaintedPathFolder.c

```diff
@@ -0,0 +1,26 @@
+#include <stdio.h>
+#include <string.h>
+
+int main(int argc, char** argv) {
+    char *userAndFile = argv[2];
+    char baseDir[PATH_MAX] = "/home/user/public/";
+    char fullPath[PATH_MAX];
+    char resolvedPath[PATH_MAX];
+
+    // Attempt to concatenate the base directory and the user-supplied path
+    snprintf(fullPath, sizeof(fullPath), "%s%s", baseDir, userAndFile);
+
+    // Resolve the absolute path, normalizing any ".." or "."
+    if (realpath(fullPath, resolvedPath) == NULL) {
```

Comment:
According to https://man7.org/linux/man-pages/man3/realpath.3.html, `realpath` should only be used with the second argument equal to `NULL`, so that it does the allocation itself (which is more robust apparently).

Reply:
:+1: